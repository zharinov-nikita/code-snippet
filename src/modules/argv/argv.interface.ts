export interface InterfaceArgv<TypeArgv> {
  format: (name: string) => TypeArgv
  find: (name: string) => TypeArgv
  getNotFormatted: () => string[]
  getFormatted: () => TypeArgv[]
}
