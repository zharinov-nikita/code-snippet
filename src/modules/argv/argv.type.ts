type Key = string | null
type Value = string | null
export interface TypeArgvGet {
  key: Key
  value: Value
}
