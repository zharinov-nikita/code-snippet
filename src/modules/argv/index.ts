export * from './argv.module'
export * from './argv.type'
