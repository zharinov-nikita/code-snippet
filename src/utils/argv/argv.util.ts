export class UtilArgv {
  public find(name: string): string {
    const args = [...process.argv.slice(2)]
    const regexp = new RegExp(`--${name}`)
    const argName = args.find((item) => (regexp.test(item) ? item : undefined))
    return argName ? argName.split('=')[1] : ''
  }
}
