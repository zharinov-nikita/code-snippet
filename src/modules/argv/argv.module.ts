import { InterfaceArgv } from './argv.interface'
import { TypeArgv } from './argv.type'

export class ModuleArgv implements InterfaceArgv<TypeArgv> {
    private args: string[]

    constructor() {
        this.args = [...process.argv.slice(2)]
    }

    public format(name: string) {
        return {
            key: name.split('=')[0].replaceAll('--', '') || null,
            value: name.split('=')[1] || null,
        }
    }

    public find(name: string) {
        const regexp = new RegExp(`${name}`)
        const arg = this.args.find((item) =>
            regexp.test(item) ? item : undefined
        )

        if (typeof arg === 'string') {
            return { ...this.format(arg) }
        }

        return {
            key: null,
            value: null,
        }
    }

    public getNotFormatted() {
        return this.args
    }

    public getFormatted() {
        return this.args.map((arg) => ({
            ...this.format(arg),
        }))
    }
}
