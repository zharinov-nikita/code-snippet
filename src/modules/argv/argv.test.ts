import { describe, expect, test } from '@jest/globals'
import { ModuleArgv } from './argv.module'

describe('module/argv', () => {
  process.argv.push('--make=ford')
  const moduleArgv = new ModuleArgv()
  test('get', () => {
    //make
    expect(moduleArgv.get('make')).toStrictEqual({ key: 'make', value: 'ford' })
    expect(moduleArgv.get('make').value).toBe('ford')
    expect(moduleArgv.get('make').key).toBe('make')
    //mode
    expect(moduleArgv.get('mode')).toStrictEqual({ key: null, value: null })
    expect(moduleArgv.get('mode').value).toBeNull()
    expect(moduleArgv.get('mode').key).toBeNull()
  })
  test('snapshot', () => {
    //make
    expect(moduleArgv.get('make')).toMatchSnapshot()
    expect(moduleArgv.get('make').value).toMatchSnapshot()
    expect(moduleArgv.get('make').key).toMatchSnapshot()
    //mode
    expect(moduleArgv.get('mode')).toMatchSnapshot()
    expect(moduleArgv.get('mode').value).toMatchSnapshot()
    expect(moduleArgv.get('mode').key).toMatchSnapshot()
  })
})
