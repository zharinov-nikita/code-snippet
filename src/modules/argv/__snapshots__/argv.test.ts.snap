// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module/argv snapshot 1`] = `
{
  "key": "make",
  "value": "ford",
}
`;

exports[`module/argv snapshot 2`] = `"ford"`;

exports[`module/argv snapshot 3`] = `"make"`;

exports[`module/argv snapshot 4`] = `
{
  "key": null,
  "value": null,
}
`;

exports[`module/argv snapshot 5`] = `null`;

exports[`module/argv snapshot 6`] = `null`;
